pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -357,20 +423,59 @@ func (b *volumeBinder) findMatchingVolumes(pod *v1.Pod, claimsToBind []*bindingI
 		// Find a matching PV
 		bindingInfo.pv, err = findMatchingVolume(bindingInfo.pvc, allPVs, node, chosenPVs, true)
 		if err != nil {
-			return false, err
+			return false, nil, err
 		}
 		if bindingInfo.pv == nil {
 			glog.V(4).Infof("No matching volumes for PVC %q on node %q", getPVCName(bindingInfo.pvc), node.Name)
-			return false, nil
+			unboundClaims = append(unboundClaims, bindingInfo.pvc)
+			foundMatches = false
+			continue
 		}
 
 		// matching PV needs to be excluded so we don't select it again
 		chosenPVs[bindingInfo.pv.Name] = bindingInfo.pv
+		preBoundClaims = append(preBoundClaims, bindingInfo)
 	}
 
 	// Mark cache with all the matches for each PVC for this node
-	b.podBindingCache.UpdateBindings(pod, node.Name, claimsToBind)
-	glog.V(4).Infof("Found matching volumes on node %q", node.Name)
+	if len(preBoundClaims) > 0 {
+		b.podBindingCache.UpdateBindings(pod, node.Name, preBoundClaims)
+	}
+
+	if foundMatches {
+		glog.V(4).Infof("Found matching volumes for all unbound claims on node %q", node.Name)
+	}
+
+	return
+}
+
+func (b *volumeBinder) checkVolumeProvisions(pod *v1.Pod, claimsToProvision []*v1.PersistentVolumeClaim, node *v1.Node) (provisionSatisfied bool, err error) {
+	provisionedClaims := []*v1.PersistentVolumeClaim{}
+
+	for _, claim := range claimsToProvision {
+		className := v1helper.GetPersistentVolumeClaimClass(claim)
+		if className == "" {
+			return false, fmt.Errorf("no class for claim %q", getPVCName(claim))
+		}
+
+		// TODO: Check if the node can satisfy the topology requirement in the class
+
+		// TODO: Check if capacity of the node domain in the storage class
+		// can satisfy resource requirement of given claim
+
+		// The claims from method args can be pointing to watcher cache. We must not
+		// modify these, therefore create a copy.
+		// TODO: move to assume phase once we find way to get topology info in AssumePodVolumes()
```

Comment:
can you explain what the issue is?  AssumePodVolumes does pass in the node name

Reply:
It's mainly for the "provisionedTopology" anno, to get its value, we'll need to fetch the storageclass and the node. And I don't want to repeat it in assume phase again.

At first I implement the plumbing. allowedTopology and capacity report part at one time, and comments here is not updated accordingly after prs slit

If we finally decide not to get the capacity part in, I'll consider to move the part to assume phase.